src/lib/optimizer/strategy/disjunction_to_union_rule.cpp

```diff
@@ -0,0 +1,63 @@
+#include "disjunction_to_union_rule.hpp"
+
+#include <expression/expression_utils.hpp>
+
+#include "expression/logical_expression.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+
+namespace opossum {
+
+std::string DisjunctionToUnionRule::name() const { return "Disjunction to Union Rule"; }
+
+void DisjunctionToUnionRule::apply_to(const std::shared_ptr<AbstractLQPNode>& root) const {
+  Assert(root->type == LQPNodeType::Root, "DisjunctionToUnionRule needs root to hold onto");
+
+  /**
+   * Step 1:
+   *    - Collect PredicateNodes that can be split up into multiple ones into `predicate_nodes_to_flat_disjunctions`
```

Comment:
I understand why `ExpressionReductionRule` is not a good place for this.

As this is c/p from the PredicateSplitUpRule, have you considered joining those two rules? Not only would that reduce the code duplication and the number of LQP traversals, but it might also enable us to handle nested and/ors.

Reply:
As discussed yesterday, this is not possible until #1813 is fixed.

Reply:
There also is another conflict with the `JoinOrderingRule`, which needs to run before the `DisjunctionToUnionRule`.